### PR TITLE
Improve invalid proxy error message

### PR DESCRIPTION
**Pull request by bronzdoc**
# Description:

closes https://github.com/rubygems/rubygems/issues/1783

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: lib/rubygems/local_remote_options.rb [hunk 1]
```diff
@@ -24,8 +25,13 @@ def accept_uri_http
         raise OptionParser::InvalidArgument, value
       end
 
-      unless ['http', 'https', 'file', 's3'].include?(uri.scheme)
-         raise OptionParser::InvalidArgument, value
+      valid_uri_schemes = ["http", "https", "file", "s3"]
+      unless valid_uri_schemes.include?(uri.scheme)
+        msg =  "Invalid uri scheme for #{value}\nPreface URLs with one of #{valid_uri_schemes.map{|s| "#{s}://"}}"
+
+        ui.say(msg)
```
Comment (segiddins) [inline lib/rubygems/local_remote_options.rb]: is using `ui` preferable to appending to the `reason` in the `OptionParser::InvalidArgument` error?

Comment (bronzdoc) [reply lib/rubygems/local_remote_options.rb]: The OpttionParser.accept is overwriting the whatever y pass as a reason to `OptionParser::InvalidArgument` with the argument is passed to the block, in this case `value`
```
    OptionParser.accept URI::HTTP do |value| ...
```

I don't know if there's a way to make OptionParser.accept to stop doing that, so i came with that solution, i will investigate a little more in the evenening

Comment (bronzdoc) [reply lib/rubygems/local_remote_options.rb]: @segiddins one option would be to throw a different error, the problem i described only happens when we raise `OptionParser::InvalidArgument ` inside the `OptionParser.accept`, what you think?

Comment (segiddins) [reply lib/rubygems/local_remote_options.rb]: I'm 👍 for throwing another error

**Comment by bronzdoc** [comment]
PR updated @segiddins

**Comment by segiddins** [comment]
Awesome! @bundlerbot r+

**Pull request merged**